third_party/amd/lib/TritonAMDGPUTransforms/RefineOps.cpp

```diff
@@ -636,6 +636,60 @@ LogicalResult rewriteLocalStoreOp(OpBuilder &rewriter,
   return success();
 }
 
+// Reduce ops have different intput and output shapes and produce
+// sliced layouts.
+// This currently only supports 2d inputs.
+LogicalResult rewriteReduceOp(OpBuilder &rewriter, triton::ReduceOp op) {
+  auto ctx = op->getContext();
+  auto loc = op.getLoc();
+  uint32_t axisReduce = op.getAxis();
+  uint32_t axisNonReduce = (axisReduce+1)%2;
+
+  // Calculate refined shape.
+  auto src = op->getOperand(0);
```

Comment:
You can check the number of operands and fail the rewrite if the number of operands is more than `1`